### ISSUE TITLE
70.75 % mAP on testing data of DIOR dataset

### ISSUE THREAD

**Issue by Ismaildata** [Shank2358/NPMMR-Det#20]
Hi sir , please i trained your model on Dior dataset using the same Train.txt file which you send it to me and i got 70.75% mAP on testing data in 151 epochs , please did you use testing or validation data to test your model , if you used testing data what is the problem in my case ?

**Comment by Shank2358** [comment]
> 70.75 % mAP in dior

Hi. When only using the training set to train the model, the mAP of NPMMR-Det reported in our paper is 71.57. Due to differences in hyperparameters, batch size, etc., the reproduced results may have some errors from our reported results. Did you train our model from scratch or a pre-trained model? For this result, I recommend adjusting the hyperparameters, such as increasing the batch size or changing the weight of the learning rate, localization, and classification loss. The result of 77.38 in the paper is the result of the training model with both training and validation sets.
In addition, there are some possible factors, the first is that the size of the anchor needs to be re-adjusted (the k-means algorithm of YOLO) if the training set and the validation set are used to train the model. Second, the 'stadium' and 'ground track field' may have the problem that the centers of the bounding boxes overlap. This will lead to the situation that the label assignment can only be assigned to one object, which needs to be alleviated by the adjustment of anchors. Third, the NMS threshold and confidence threshold will also affect the results, you can also adjust them to see how the results are.
Thank you.

**Comment by Ismaildata** [comment]
can you  please sir share me algorithm used to generate anchors , because i used Kmeans algorithm of YOLO and dosnt give same results as you .
please share me full code generate anchors using DIOR dataset .
Thank you sir in advance .

**Comment by Shank2358** [comment]
> can you please sir share me algorithm used to generate anchors , because i used Kmeans algorithm of YOLO and dosnt give same results as you . please share me full code generate anchors using DIOR dataset . Thank you sir in advance .

I used this code.
https://github.com/lars76/kmeans-anchor-boxes

**Comment by TerrafYassin** [comment]
please do you have entire code used to generate those anchors in DIOR dataset ?

**Comment by Shank2358** [comment]
> please do you have entire code used to generate those anchors in DIOR dataset ?

I used the code from the above repository, which also has examples.
Did you encounter any problems?
